### PR TITLE
Refactored: setEcomStore and setShopifyConfig actions to accept ID. (#2tzh44d)

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #

### Short Description and Why It's Useful
Refactored action setEcomStore and setShopifyConfig to accept ID as well as object, as initially the whole object was passed, posing unnecessary code. 

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/views/Settings.vue [hunk 1]
```diff
@@ -88,14 +88,13 @@ export default defineComponent({
   methods: {
     setEComStore(store: any) {
       if(this.userProfile) {
-        this.store.dispatch('user/setEcomStore', {
-          'eComStore': this.userProfile.stores.find((str: any) => str.productStoreId == store['detail'].value)
-        })
+        const productStore = this.userProfile.stores.find((str: any) => str.productStoreId === store['detail'].value);
```
Comment (adityasharma7) [inline src/views/Settings.vue:91]: We should not find the product store here, we could just pass the productStoreId here and action should handle if id is provided it finds the product store & set it and if product store is provided it should set it

File: src/views/Settings.vue [hunk 2]
```diff
@@ -88,14 +88,13 @@ export default defineComponent({
...
+        this.store.dispatch('user/setEcomStore', productStore ? productStore.productStoreId : {});
       }
     },
     setShopifyConfig(event: any){
-      const shopifyConfig = this.shopifyConfigs.find((shopifyConfig: any) => shopifyConfig.shopifyConfigId === event.detail.value)
-      this.store.dispatch('user/setCurrentShopifyConfig', shopifyConfig);
+      const currentShopifyConfig = this.shopifyConfigs.find((shopifyConfig: any) => shopifyConfig.shopifyConfigId === event.detail.value);
```
Comment (adityasharma7) [inline src/views/Settings.vue:96]: Refer comments provided for the product store

File: src/store/modules/user/actions.ts [hunk 3]
```diff
@@ -130,13 +130,15 @@ const actions: ActionTree<UserState, RootState> = {
   /**
    * update current eComStore information
    */
-  async setEcomStore({ commit, dispatch }, payload) {
+  async setEcomStore({ commit, dispatch, state }, productStore) {
+    const currentEComStore = state.currentEComStore;
+    const productStoreId = typeof productStore === 'string' ? productStore : productStore.productStoreId;
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:135]: I would suggest we check if payload has productStoreId we find the productStore and if productStore we set it as is

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Settings.vue (hunk 1 above)
Comment (k2maan) [reply src/views/Settings.vue:91]: Sure, sir, understood.

File: src/views/Settings.vue (hunk 2 above)
Comment (k2maan) [reply src/views/Settings.vue:96]: Sure, sure.

File: src/store/modules/user/actions.ts (hunk 3 above)
Comment (k2maan) [reply src/store/modules/user/actions.ts:135]: Yes, sir, will change it as stated.

File: src/views/Settings.vue [hunk 4]
```diff
@@ -88,14 +88,18 @@ export default defineComponent({
   methods: {
     setEComStore(store: any) {
       if(this.userProfile) {
+        const productStoreId = store['detail'].value
         this.store.dispatch('user/setEcomStore', {
-          'eComStore': this.userProfile.stores.find((str: any) => str.productStoreId == store['detail'].value)
+          'productStoreId' : productStoreId ? productStoreId : {}
```
Comment (adityasharma7) [inline src/views/Settings.vue:93]: Why this is needed? We could just pass the productStoreId

File: src/views/Settings.vue [hunk 5]
```diff
@@ -88,14 +88,18 @@ export default defineComponent({
...
         })
       }
     },
     setShopifyConfig(event: any){
-      const shopifyConfig = this.shopifyConfigs.find((shopifyConfig: any) => shopifyConfig.shopifyConfigId === event.detail.value)
-      this.store.dispatch('user/setCurrentShopifyConfig', shopifyConfig);
+      // const currentShopifyConfig = this.shopifyConfigs.find((shopifyConfig: any) => shopifyConfig.shopifyConfigId === event.detail.value);
+      const shopifyConfigId = event.detail.value;
+      this.store.dispatch('user/setCurrentShopifyConfig', {
+        'shopifyConfig' : shopifyConfigId ? shopifyConfigId : {}
```
Comment (adityasharma7) [inline src/views/Settings.vue:101]: Variable name should be shopifyConfigId
Why this is needed? We could just pass the shopifyConfigId

File: src/store/modules/user/actions.ts [hunk 6]
```diff
@@ -204,9 +209,14 @@ const actions: ActionTree<UserState, RootState> = {
   /**
    * update current shopify config id
    */
-  async setCurrentShopifyConfig({ commit, dispatch }, shopifyConfig) {
-    commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, shopifyConfig ? shopifyConfig : {});
-    dispatch('job/clearJobState', null, { root: true });
+  async setCurrentShopifyConfig({ commit, dispatch, state }, payload) {
+    const shopifyConfigs = state.shopifyConfigs ? state.shopifyConfigs : {};
+    let currentShopifyConfig;
+    payload.shopifyConfig.shopifyConfigId ?
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:215]: If we are getting shopifyConfig, just set it as is, if not find and set. Same applied to productStore

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts (hunk 6 above)
Comment (k2maan) [reply src/store/modules/user/actions.ts]: Made the changes, sir, please review.

File: src/store/modules/user/actions.ts [hunk 7]
```diff
@@ -130,13 +130,20 @@ const actions: ActionTree<UserState, RootState> = {
...
+  async setEcomStore({ commit, dispatch, getters }, payload) {
     dispatch('job/clearJobState', null, { root: true });
-    commit(types.USER_CURRENT_ECOM_STORE_UPDATED, payload.eComStore);
-    dispatch('getShopifyConfig', payload.eComStore.productStoreId);
+    let productStore;
+
+    if(payload.productStore) {
+      commit(types.USER_CURRENT_ECOM_STORE_UPDATED, payload.productStore);
+    } else {
+      productStore = getters.getUserProfile.stores.find((store: any) => store.productStoreId === payload.productStoreId);
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:140]: No need to use getter, I think we could directly access the state

File: src/store/modules/user/actions.ts [hunk 8]
```diff
@@ -130,13 +130,20 @@ const actions: ActionTree<UserState, RootState> = {
...
    */
-  async setEcomStore({ commit, dispatch }, payload) {
+  async setEcomStore({ commit, dispatch, getters }, payload) {
     dispatch('job/clearJobState', null, { root: true });
-    commit(types.USER_CURRENT_ECOM_STORE_UPDATED, payload.eComStore);
-    dispatch('getShopifyConfig', payload.eComStore.productStoreId);
+    let productStore;
+
+    if(payload.productStore) {
+      commit(types.USER_CURRENT_ECOM_STORE_UPDATED, payload.productStore);
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:138]: We should commit in both the cases, here we should assign productStore with payload.productStore

File: src/store/modules/user/actions.ts [hunk 9]
```diff
@@ -130,13 +130,20 @@ const actions: ActionTree<UserState, RootState> = {
...
+    } else {
+      productStore = getters.getUserProfile.stores.find((store: any) => store.productStoreId === payload.productStoreId);
+    }
+    
+    dispatch('getShopifyConfig', productStore ? productStore.productStoreId : payload.productStore.productStoreId);
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:143]: Once we have `productStore` set for both the cases, we will not need this ternary operation we could use `productStore.productStoreId` for both the cases

File: src/store/modules/user/actions.ts [hunk 10]
```diff
@@ -130,13 +130,20 @@ const actions: ActionTree<UserState, RootState> = {
...
     await UserService.setUserPreference({
       'userPrefTypeId': 'SELECTED_BRAND',
-      'userPrefValue': payload.eComStore.productStoreId
+      'userPrefValue': productStore ? productStore.productStoreId : payload.productStore.productStoreId
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:146]: Same here. refer my previous comment

File: src/store/modules/user/actions.ts [hunk 11]
```diff
@@ -204,9 +211,16 @@ const actions: ActionTree<UserState, RootState> = {
...
-    commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, shopifyConfig ? shopifyConfig : {});
-    dispatch('job/clearJobState', null, { root: true });
+  async setCurrentShopifyConfig({ commit, dispatch, state }, payload) {
+    if(payload.shopifyConfig) {
+      commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, payload.shopifyConfig);
+    } else {
+      const shopifyConfigs = state.shopifyConfigs;
+      const currentShopifyConfig = shopifyConfigs.find((configs: any) => configs.shopifyConfigId === payload.shopifyConfigId)
+      commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, currentShopifyConfig);
+    }
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:221]: I would suggest following code for better readability
```suggestion
    let currentShopifyConfig = payload.shopifyConfig;
    if(!currentShopifyConfig) {
      currentShopifyConfig = state.shopifyConfigs.find((shopifyConfig: any) => shopifyConfig.shopifyConfigId === payload.shopifyConfigId)
    } 
    commit(types.USER_CURRENT_SHOPIFY_CONFIG_UPDATED, currentShopifyConfig ? currentShopifyConfig : {});
```

File: src/views/Settings.vue [hunk 12]
```diff
@@ -88,14 +88,14 @@ export default defineComponent({
   methods: {
     setEComStore(store: any) {
       if(this.userProfile) {
-        this.store.dispatch('user/setEcomStore', {
-          'eComStore': this.userProfile.stores.find((str: any) => str.productStoreId == store['detail'].value)
-        })
+        const productStoreId = store['detail'].value
+        this.store.dispatch('user/setEcomStore', { 'productStoreId': productStoreId })
```
Comment (adityasharma7) [inline src/views/Settings.vue:92]: Why not?
```suggestion
        this.store.dispatch('user/setEcomStore', { 'productStoreId': store['detail'].value })
```

Comment (adityasharma7) [reply src/views/Settings.vue:92]: Also, we could rename store to event like `setShopifyConfig()` method and access directly event.detail.value

File: src/views/Settings.vue [hunk 13]
```diff
@@ -88,14 +88,14 @@ export default defineComponent({
...
       }
     },
     setShopifyConfig(event: any){
-      const shopifyConfig = this.shopifyConfigs.find((shopifyConfig: any) => shopifyConfig.shopifyConfigId === event.detail.value)
-      this.store.dispatch('user/setCurrentShopifyConfig', shopifyConfig);
+      // const currentShopifyConfig = this.shopifyConfigs.find((shopifyConfig: any) => shopifyConfig.shopifyConfigId === event.detail.value);
```
Comment (adityasharma7) [inline src/views/Settings.vue:96]: Remove unwanted code

File: src/views/Settings.vue [hunk 14]
```diff
@@ -88,14 +88,14 @@ export default defineComponent({
...
+      const shopifyConfigId = event.detail.value;
+      this.store.dispatch('user/setCurrentShopifyConfig', { 'shopifyConfigId': shopifyConfigId });
```
Comment (adityasharma7) [inline src/views/Settings.vue:98]: Why not?
```suggestion
      this.store.dispatch('user/setCurrentShopifyConfig', { 'shopifyConfigId': event.detail.value });
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Settings.vue (hunk 14 above)
Comment (k2maan) [reply src/views/Settings.vue]: Implemented the changes, sir, please review.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**